kubernetes/windows/main.ps1

```
@@ -289,35 +289,99 @@ function Start-Fluent {
+
+    # Setting delay auto start for telegraf since there have been known issues with windows server and telegraf -
+    # https://github.com/influxdata/telegraf/issues/4081
+    # https://github.com/influxdata/telegraf/issues/3601
     try {
-        # This is required when the root CA certs are different for some clouds.
-        $caCerts=Invoke-WebRequest 'http://168.63.129.16/machine?comp=acmspackage&type=cacertificates&ext=json' -UseBasicParsing | ConvertFrom-Json
-        if (![string]::IsNullOrEmpty($caCerts)) {
-            $certificates = $caCerts.Certificates
-            for ($index = 0; $index -lt $certificates.Length ; $index++) {
-                $name=$certificates[$index].Name
-                $certificates[$index].CertBody > $name
-                Write-Host "name: $($name)"
-                Import-Certificate -FilePath .\$name  -CertStoreLocation 'Cert:\LocalMachine\Root' -Verbose
-            }
+        $serverName = [System.Environment]::GetEnvironmentVariable("PODNAME", "process")
+        if (![string]::IsNullOrEmpty($serverName)) {
+            sc.exe \\$serverName config telegraf start= delayed-auto
+            Write-Host "Successfully set delayed start for telegraf"
+
+        } else {
+            Write-Host "Failed to get environment variable PODNAME to set delayed telegraf start"
         }
     }
     catch {
-        $e = $_.Exception
-        Write-Host $e
-        Write-Host "exception occured in Bootstrap-CACertificates..."
+            $e = $_.Exception
+            Write-Host $e
+            Write-Host "exception occured in delayed telegraf start.. continuing without exiting"
     }
+    Write-Host "Running telegraf service in test mode"
+    C:\opt\telegraf\telegraf.exe --config "C:\etc\telegraf\telegraf.conf" --test
+    Write-Host "Starting telegraf service"
+    C:\opt\telegraf\telegraf.exe --service start
+
+    # Trying to start telegraf again if it did not start due to fluent bit not being ready at startup
```

if telegraf failed to start, will throw exception or silently fail?